server/src/main/java/org/apache/lucene/codecs/experimental/Lucene92CustomCodec.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.apache.lucene.codecs.experimental;
+
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.StoredFieldsFormat;
+import org.apache.lucene.codecs.lucene92.Lucene92Codec;
+
+/** Custom codec for different compression algorithm */
+public class Lucene92CustomCodec extends FilterCodec {
+
+    private final StoredFieldsFormat storedFieldsFormat;
+
+    public static final int defaultCompressionLevel = 6;
```

Comment:
for `final static` better give names like `DEFAULT_COMPRESSION_LEVEL`

Reply:
Fixed.